app/monitor.js

```diff
@@ -15,7 +15,25 @@ var async = require('async'),
                 .digest('hex');
         }
     },
-    sortBuilds = function (newBuilds, sortOrder) {
+    sortBuilds = function (newBuilds, sortOrder, errorFirst) {
+        if (errorFirst)
+        {
+            var errorChunk = [];
+            var normalChunk = [];
+            for (var i = 0; i < newBuilds.length; i++)
+            {
+                if (newBuilds[i].hasErrors)
+                    errorChunk.push(newBuilds[i]);
+                else
+                    normalChunk.push(newBuilds[i]);
+            }
+
+            sortBuilds(errorChunk, sortOrder);
+            sortBuilds(normalChunk, sortOrder);
+
+            return errorChunk.concat(normalChunk);
```

Comment:
In that way, the `sortOrder` configuration will be ignored. This behavior is very complicated for a user to understand. I think the new sorting code could be moved after the existing sorting-option. So, `newBuilds` will correctly be sorted by date or project, and then the erroneous builds will be moved to the beginning (if configured).

Reply:
I've moved it to the end of the function.
Actually, sortOrder was not ignored, the old functionality was not returning the value just sorting it, but you're right, this is confusing to use it like that.